ecosystem/sep-0012.md

```diff
@@ -363,3 +365,104 @@ PUT [KYC_SERVER || TRANSFER_SERVER]/customer/callback
 | Success                                   | `200 OK`           |
 | Client not authenticated properly         | `401 Unauthorized` |
 | Anchor has no information on the customer | `404 Not Found`    |
+
+## Customer Files
+
+Passing binary fields such as `photo_id_front` or `organization.photo_proof_address` in [PUT /customer](#customer-put) requests must be done using the `multipart/form-data` content type. This is acceptable in most cases, but `multipart/form-data` does not support nested data structures such as arrays or sub-objects.
+
+This endpoint is intended to decouple requests containing binary fields from requests containing nested data structures, supported by content types such as `application/json`. This endpoint is optional and only needs to be supported if your use case requires accepting nested data structures in `PUT /customer` requests.
+
+Once a file has been uploaded using this endpoint, it's `id` can be used in subsequent `PUT /customer` requests. The field name for the `id` should be the appropriate [SEP-9](sep-0009.md) field followed by `_file_id`. For example, if `file_abc` is returned as an `id` from [POST /customer/files](post-request), it can be used in a `PUT /customer` request like so:
+
+```json
+{
+   "account":"GBORFR3GDNVZ5PLUTBDQHKGWVD26CQUHORO2T3SDQ2JPLGLUJCCA5GK6",
+   "memo":"21bf91a4-7db1-401d-8108-fab7660a45d6",
+   "memo_type":"text",
```

Comment:
Its possible to provide an `id` instead of `account`/`memo`/`memo_type`, so I can add an example using it. 

Using `id` means customer data was already provided earlier, so it would probably only be used when a file didn't pass validation and needs to be updated for an existing customer.

Reply:
This may get confusing because the `id` in the response will be the file not the customer.

I have a different thought after thinking about this more. A customer should always exist when files are being uploaded because files are uploaded for verification, and verification is only requested after the provision of an initial data set.

If the customer ID is always known this would improve the shape of this API. For example the /customer/files endpoint could become `/customer/{customer-id}/files`.

Reply:
Although that would be inconsistent with the other endpoints unfortunately, since PUT and GET for customers don't use the customer ID in the URL path, nor does the customer callback.

Reply:
> A customer should always exist when files are being uploaded because files are uploaded for verification, and verification is only requested after the provision of an initial data set.

There are several SEP-9 image fields that can be sent in the initial `PUT /customer` request, like `photo_id_front`, `photo_proof_residence`. I think you're getting mixed up with the new `PUT /customer/verification` endpoint, which handles the case where verification can only happen after the associated field was provided.

Reply:
I was getting confused with that, thanks. I see your point.

For consistency with the callback endpoint `id` in the request makes sense for the user, maybe we should change the response `id` to be `file_id`? Do you have any ideas for dealing with the symmetry?

Reply:
I think using `file_id` would be fine. It matches the suffix of the SEP-9 attributes needed to indicate that the value is a file id.